src/makeInfinityAdder.js

```diff
@@ -1,12 +1,23 @@
 'use strict';
 
-/***
- *
- * @return {function}
- */
-
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (num) => {
+    if (typeof num === 'number') {
+      sum += num;
+
+      return adder;
+    } else {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
```

Comment:
```suggestion
    } 
    
    const result = sum;

    sum = 0;

    return result;
```
avoid else statement where it possible